platform_plugin_forum_email_notifier/tests/test_tasks.py

```diff
@@ -0,0 +1,527 @@
+
+
+class TestSendEmailNotification(TestCase):
+    """Unit test for `send_email_notification` task."""
+
+    get_mock = patch(f"{module_path}.User.objects.get")
+    get_course_overview_or_none_mock = patch(
+        f"{module_path}.get_course_overview_or_none"
+    )
+    get_user_preference_mock = patch(f"{module_path}.get_user_preference")
+    send_forum_email_notification_mock = patch(
+        f"{module_path}.send_forum_email_notification"
+    )
+
+    def setUp(self) -> None:
+        """
+        Set up common test data for each test case.
+        """
+        self.send_email_notification_args = {
+            "thread_id": "test-thread-id",
+            "discussion": None,
+            "course_id": "test-course-id",
+            "body": "<p>test-body<p>",
+            "title": "test-title",
+            "url": "test-url/",
+            "author_id": 1,
+            "author_username": "test-author-username",
+            "author_email": "test@author-email.com",
+            "object_type": ForumObject.THREAD,
+            "subscriber": 1,
+            "context": {},
+        }
+
+    @get_mock
+    @get_course_overview_or_none_mock
+    @get_user_preference_mock
+    @send_forum_email_notification_mock
+    def test_send_email_notification_user_exists(
+        self,
+        mock_send_forum_email_notification: Mock,
+        mock_get_user_preference: Mock,
+        mock_get_course_overview_or_none: Mock,
+        mock_get: Mock,
+    ):
+        """
+        Check `send_email_notification` when user exists.
+
+        Expected result:
+            - Send an email notification when user exists.
+        """
+        mock_user = Mock(spec=User, id=1, email="test@user-email.com")
+        mock_get.return_value = mock_user
+        mock_get_course_overview_or_none.return_value = Mock()
+        mock_get_user_preference.return_value = "en"
+
+        send_email_notification(**self.send_email_notification_args)
+
+        mock_send_forum_email_notification.assert_called_once_with(
+            recipient=Recipient(1, "test@user-email.com"),
+            language="en",
+            user_context={
+                "user": mock_user,
+                "course_id": "test-course-id",
+                "course_name": mock_get_course_overview_or_none.return_value.display_name,
+                "thread_id": "test-thread-id",
+                "discussion": None,
+                "body": "test-body",
+                "title": "test-title",
+                "url": "test-url/discussions/test-course-id/posts/test-thread-id",
+                "author_id": 1,
+                "author_username": "test-author-username",
+                "author_email": "test@author-email.com",
+                "object_type": ForumObject.THREAD,
+            },
+        )
+
+    @get_mock
+    def test_send_email_notification_user_does_not_exist(self, mock_get: Mock):
+        """
+        Check `send_email_notification` when user does not exist.
```

Comment:
If we're gonna use `Check...` let's add `Check send_email_notification behavior...`